program/src/processor.rs

```diff
@@ -977,36 +983,77 @@ pub fn process(program_id: &Pubkey, accounts: &[AccountInfo], input: &[u8]) -> P
+            amount,
+            validator_index,
+        } => process_withdraw(program_id, amount, validator_index, accounts),
         LidoInstruction::ChangeRewardDistribution {
             new_reward_distribution,
         } => process_change_reward_distribution(program_id, new_reward_distribution, accounts),
         LidoInstruction::AddValidatorV2 => process_add_validator(program_id, accounts),
-        LidoInstruction::RemoveValidator => process_remove_validator(program_id, accounts),
-        LidoInstruction::DeactivateValidator => process_deactivate_validator(program_id, accounts),
-        LidoInstruction::AddMaintainer => process_add_maintainer(program_id, accounts),
-        LidoInstruction::RemoveMaintainer => process_remove_maintainer(program_id, accounts),
-        LidoInstruction::MergeStake => process_merge_stake(program_id, accounts),
-        LidoInstruction::DeactivateValidatorIfCommissionExceedsMax => {
-            process_deactivate_validator_if_commission_exceeds_max(program_id, accounts)
+        LidoInstruction::RemoveValidatorV2 { validator_index } => {
+            process_remove_validator(program_id, validator_index, accounts)
+        }
+        LidoInstruction::DeactivateValidatorV2 { validator_index } => {
+            process_deactivate_validator(program_id, validator_index, accounts)
+        }
+        LidoInstruction::AddMaintainerV2 => process_add_maintainer(program_id, accounts),
+        LidoInstruction::RemoveMaintainerV2 { maintainer_index } => {
+            process_remove_maintainer(program_id, maintainer_index, accounts)
+        }
+        LidoInstruction::MergeStakeV2 { validator_index } => {
+            process_merge_stake(program_id, validator_index, accounts)
+        }
+        LidoInstruction::DeactivateValidatorIfCommissionExceedsMax { validator_index } => {
+            process_deactivate_validator_if_commission_exceeds_max(
+                program_id,
+                validator_index,
+                accounts,
+            )
         }
         LidoInstruction::SetMaxValidationCommission {
             max_commission_percentage,
         } => process_set_max_commission_percentage(program_id, max_commission_percentage, accounts),
+        LidoInstruction::WithdrawInactiveStake
+        | LidoInstruction::CollectValidatorFee
+        | LidoInstruction::ClaimValidatorFee
+        | LidoInstruction::MergeStake
+        | LidoInstruction::AddValidator
+        | LidoInstruction::AddMaintainer
+        | LidoInstruction::DeactivateValidator
+        | LidoInstruction::RemoveMaintainer
+        | LidoInstruction::RemoveValidator
+        | LidoInstruction::StakeDeposit { .. }
+        | LidoInstruction::Unstake { .. }
+        | LidoInstruction::Withdraw { .. } => {
+            msg!("{:?} is no longer supported since v2. Please check the changelog in the repository for update instructions.", instruction);
+            Err(LidoError::InstructionIsDeprecated.into())
```

Comment:
Can you add the newly deprecated functions to the changelog as well? The main thing to explain is that many of these functions now take the validator and maintainer index as transaction data, to make these instructions more efficient, so clients need to compute that index, similar to pre-computing a bump seed so it only has to be verified on-chain.

Reply:
done